pkg/operators/v1alpha1/catalogsource_types.go

```diff
@@ -96,7 +100,30 @@ type RegistryPoll struct {
 	// Interval is used to determine the time interval between checks of the latest catalog source version.
 	// The catalog operator polls to see if a new version of the catalog source is available.
 	// If available, the latest image is pulled and gRPC traffic is directed to the latest catalog source.
-	Interval *metav1.Duration `json:"interval,omitempty"`
+	Interval     *metav1.Duration `json:"interval,omitempty"`
+	ParsingError string           `json:"-"`
+}
+
+// UnmarshalJSON implements the encoding/json.Unmarshaler interface.
+func (u *UpdateStrategy) UnmarshalJSON(data []byte) (err error) {
+	var updateStrategyMap map[string]map[string]string
+	if err = json.Unmarshal(data, &updateStrategyMap); err != nil {
+		return err
+	}
+	registryPoll := &RegistryPoll{}
```

Comment:
Just a quick thought (and this can be ignored): it looks like you can avoid an if/else statement structure by defaulting the `Interval` field to the `defaultTime` duration, and only override that value when the parsing the duration provided doesn't provide a parsing ignore.

Reply:
Sure, but you'd also have to do everything around the default time (parse the `15m` string to `time.Duration`, format the error etc) by default too, even if the error is nil. Having the if/else seems cleaner and avoids unnecessary operations every time.